File: src/stores/health.ts
import { defineStore } from 'pinia'
import { ref } from 'vue'
import { healthApi } from '@/api'

export interface ExerciseLog {
  id: number
  type: string
  duration: number
  calories: number
  distance: number
  steps: number
  note: string
  date: string | null
  created_at: string
}

export interface SleepLog {
  id: number
  bed_time: string | null
  wake_time: string | null
  duration: number
  quality: number
  note: string
  date: string | null
  created_at: string
}

export interface MoodLog {
  id: number
  mood: number
  stress_level: number
  note: string
  tags: string[]
  date: string | null
  created_at: string
}

export const useHealthStore = defineStore('health', () => {
  const exerciseLogs = ref<ExerciseLog[]>([])
  const sleepLogs = ref<SleepLog[]>([])
  const moodLogs = ref<MoodLog[]>([])
  const loading = ref(false)

  async function fetchAll() {
    loading.value = true
    try {
      const [exRes, slRes, moRes] = await Promise.allSettled([
        healthApi.getExerciseLogs(),
        healthApi.getSleepLogs(),
        healthApi.getMoodLogs(),
      ])
      if (exRes.status === 'fulfilled') exerciseLogs.value = exRes.value.data
      if (slRes.status === 'fulfilled') sleepLogs.value = slRes.value.data
      if (moRes.status === 'fulfilled') moodLogs.value = moRes.value.data
    } finally { loading.value = false }
  }

  async function addExercise(log: Partial<ExerciseLog>) {
    const { data } = await healthApi.createExerciseLog(log)
    exerciseLogs.value.unshift(data)
    return data
  }

  async function addSleep(log: Partial<SleepLog>) {
    const { data } = await healthApi.createSleepLog(log)
    sleepLogs.value.unshift(data)
    return data
  }

  async function addMood(log: Partial<MoodLog>) {
    const { data } = await healthApi.createMoodLog(log)
    moodLogs.value.unshift(data)
    return data
  }

  return {
    exerciseLogs, sleepLogs, moodLogs, loading,
    fetchAll, addExercise, addSleep, addMood,
  }
})

File: server/app/models/health.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey, JSON, Text, Float
from datetime import datetime, timezone
from app.database import Base


class ExerciseLog(Base):
    __tablename__ = "exercise_logs"
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    type = Column(String(50), nullable=False)  # running/walking/cycling/gym/...
    duration = Column(Integer, default=0)  # minutes
    calories = Column(Float, default=0)
    distance = Column(Float, default=0)  # km
    steps = Column(Integer, default=0)
    note = Column(Text, default="")
    date = Column(DateTime, nullable=True)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))


class SleepLog(Base):
    __tablename__ = "sleep_logs"
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    bed_time = Column(DateTime, nullable=True)
    wake_time = Column(DateTime, nullable=True)
    duration = Column(Float, default=0)  # hours
    quality = Column(Integer, default=3)  # 1-5
    note = Column(Text, default="")
    date = Column(DateTime, nullable=True)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))


class MoodLog(Base):
    __tablename__ = "mood_logs"
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    mood = Column(Integer, default=3)  # 1-5 (very bad to very good)
    stress_level = Column(Integer, default=3)  # 1-5
    note = Column(Text, default="")
    tags = Column(JSON, default=list)
    date = Column(DateTime, nullable=True)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))
